api/src/kots_app/resolvers/kots_app_mutations.ts

```diff
@@ -228,14 +228,20 @@ export function KotsMutations(stores: Stores) {
           /* no need to handle, rbac problem or not a path we can read registry */
         }
 
-        return {
-          hasPreflight: kotsApp.hasPreflight,
-          isAirgap: parsedLicense.spec.isAirgapSupported,
-          needsRegistry,
-          slug: kotsApp.slug,
-          isConfigurable: kotsApp.isAppConfigurable()
+        if (kotsApp) {
+          return {
+            hasPreflight: kotsApp.hasPreflight,
+            isAirgap: parsedLicense.spec.isAirgapSupported,
+            needsRegistry,
+            slug: kotsApp.slug,
+            isConfigurable: kotsApp.isAppConfigurable()
+          }
+        }
+        else {
```

Comment:
no need to handle this case if `kotsAppFromLicenseData` throws an error instead of returning undefined